WireDomain/Sources/WireDomain/Repositories/FeatureConfigRepository.swift

```diff
@@ -0,0 +1,352 @@
+    /// - Returns: `true` if user should be notified.
+
+    func fetchNeedsToNotifyUser(forFeatureName name: Feature.Name) async throws -> Bool
+
+    /// Stores a flag indicating whether the user should be notified of a given feature.
+    /// - Parameter notifyUser: Whether the user should be notified for a given feature.
+    /// - Parameter name: The name of the feature to set the flag for.
+
+    func storeNeedsToNotifyUser(_ notifyUser: Bool, forFeatureName name: Feature.Name) async throws
+
+}
+
+final class FeatureConfigRepository: FeatureConfigRepositoryProtocol {
+
+    // MARK: - Properties
+
+    private let featureConfigsAPI: any FeatureConfigsAPI
+    private let context: NSManagedObjectContext
+
+    // MARK: - Object lifecycle
+
+    init(
+        featureConfigsAPI: any FeatureConfigsAPI,
+        context: NSManagedObjectContext
+    ) {
+        self.featureConfigsAPI = featureConfigsAPI
+        self.context = context
+    }
+
+    // MARK: - Public
+
+    func pullFeatureConfigs() -> AsyncThrowingStream<FeatureState, Error> {
+        AsyncThrowingStream<FeatureState, Error> { continuation in
+            Task {
+                do {
+                    let featureConfigs = try await featureConfigsAPI.getFeatureConfigs()
+
+                    for featureConfig in featureConfigs {
+                        do {
+                            try await storeFeatureConfig(featureConfig)
+                            if let featureState = try await getFeatureState(forFeatureConfig: featureConfig) {
+                                continuation.yield(featureState)
+                            }
+                        } catch {
+                            continuation.finish(throwing: FeatureConfigRepositoryError.failedToStoreConfigLocally(error))
+                        }
+                    }
+
+                    continuation.finish()
+                } catch {
+                    continuation.finish(throwing: error)
+                }
+            }
+        }
+    }
+
+    func fetchFeatureConfig<T: Decodable>(withName name: Feature.Name, type: T.Type) async throws -> LocalFeature<T> {
+        try await context.perform { [context] in
+            guard let feature = Feature.fetch(name: name, context: context) else {
+                throw FeatureConfigRepositoryError.failedToFetchConfigLocally
+            }
+
+            if let config = feature.config {
+                let decoder = JSONDecoder()
+                let config = try decoder.decode(type, from: config)
+
+                return LocalFeature(status: feature.status, config: config)
+            }
+
+            return LocalFeature(status: feature.status, config: nil)
+        }
+    }
+
+    func fetchNeedsToNotifyUser(forFeatureName name: Feature.Name) async throws -> Bool {
+        await context.perform { [context] in
+            let feature = Feature.fetch(name: name, context: context)
+            return feature?.needsToNotifyUser ?? false
```

Comment:
I'm not what should happen here but it feels strange to fallback to `false`. Perhaps it should throw? Or return optional `Bool`?

Is is a correct behavior to check this when the Feature hasn't been persisted?

Reply:
It was designed like this in legacy codebase so I'm guessing if we fail fetching the `needsToNotifyUser` we fallback to false and don't let the user know anything about it. But since the repository should not be responsible of handling that kind of logic, we should just throw here so I'll make the change, thanks